lib/ansible/cli/doc.py

```diff
@@ -328,6 +586,30 @@ def run(self):
                 docs = DocCLI._list_keywords()
             else:
                 docs = DocCLI._get_keywords_docs(context.CLIARGS['args'])
+        elif plugin_type == 'role':
+            if context.CLIARGS['list_dir']:
+                # If an argument was given with --list, it is a collection filter
+                coll_filter = None
+                if len(context.CLIARGS['args']) == 1:
+                    coll_filter = context.CLIARGS['args'][0]
+                    if not AnsibleCollectionRef.is_valid_collection_name(coll_filter):
```

Comment:
This prevents listing standalone roles, such as `caddy` or `docker`. Is that intentional?

Reply:
```
> ansible-doc -t role --list caddy
ERROR! Invalid collection name (must be of the form namespace.collection): caddy
```

Reply:
Yes, that was intentional. Firstly, it mirrors existing `--list` behavior for other plugin types supported by `ansible-doc` where only a collection filter is supported. Secondly, it seems rather pointless to list a single role whose name we already know. Is there a use here I'm not considering?

Reply:
Ok, I understand now. I hadn't ever used this with other plugin types before. It's effectively a way to list all plugin types within a collection. And now this can list all roles within a collection. Makes sense.